lib/app/features/chat/providers/private_direct_chat_provider.c.dart

```diff
@@ -0,0 +1,370 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'dart:convert';
+import 'dart:developer';
+import 'dart:io';
+import 'dart:isolate';
+
+import 'package:cryptography/cryptography.dart';
+import 'package:ion/app/exceptions/exceptions.dart';
+import 'package:ion/app/features/chat/model/entities/private_direct_message_data.c.dart';
+import 'package:ion/app/features/chat/model/entities/private_message_reaction_data.c.dart';
+import 'package:ion/app/features/core/model/media_type.dart';
+import 'package:ion/app/features/core/providers/env_provider.c.dart';
+import 'package:ion/app/features/nostr/model/entity_expiration.c.dart';
+import 'package:ion/app/features/nostr/model/file_alt.dart';
+import 'package:ion/app/features/nostr/model/nostr_entity.dart';
+import 'package:ion/app/features/nostr/providers/nostr_event_signer_provider.c.dart';
+import 'package:ion/app/features/nostr/providers/nostr_notifier.c.dart';
+import 'package:ion/app/features/nostr/providers/nostr_upload_notifier.c.dart';
+import 'package:ion/app/services/compressor/compress_service.c.dart';
+import 'package:ion/app/services/media_service/media_service.c.dart';
+import 'package:ion/app/services/nostr/ion_connect_gift_wrap_service.dart';
+import 'package:ion/app/services/nostr/ion_connect_seal_service.dart';
+import 'package:nostr_dart/nostr_dart.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'private_direct_chat_provider.c.g.dart';
+
+@riverpod
+class PrivateDirectChatProvider extends _$PrivateDirectChatProvider {
+  @override
+  void build() {}
+
+  Future<List<NostrEntity?>> sentMessage({
+    required String content,
+    required List<String> participantsPubkeys,
+    String? subject,
+    List<MediaFile> mediaFiles = const [],
+  }) async {
+    final sealService = IonConnectSealServiceImpl();
```

Comment:
Let's inject`IonConnectSealServiceImpl` and `IonConnectGiftWrapServiceImpl` via the providers

Reply:
Done